Include/pymacro.h

```diff
@@ -15,10 +15,10 @@
 // MSVC makes static_assert a keyword in C11-17, contrary to the standards.
 //
 // In C++11 and C2x, static_assert is a keyword, redefining is undefined
-// behaviour. So only define if building as C (if __STDC_VERSION__ is defined),
-// not C++, and only for C11-17.
+// behaviour. So only define if building as C, not C++ (if __cplusplus is
+// not defined), and only for C11-17.
 #if !defined(static_assert) && (defined(__GNUC__) || defined(__clang__)) \
-     && defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L \
+     && !defined(__cplusplus) && __STDC_VERSION__ >= 201112L \
```

Comment:
I'd keep `defined(__STDC_VERSION__)` and add `&& !defined(__cplusplus)` to the end. I believe it was there for some reason, e.g. GCC and Clang may warn undefined identifiers(`-Wundef`).

Reply:
You are right, better be safe here. I kept the `defined(__STDC_VERSION__)` in the check.